jquery.flot.legend.js

```diff
@@ -23,7 +23,7 @@
     function insertLegend(plot) {
```

Comment:
This function is manipulating the DOM which makes it really slow. Please make sure there is a check which ensures this function is not called if it's not necessary.

One possibility to do this:
* move the code that computes the **entries** object in a different function
* each time **setupGrid()** is called compute the **entries** again and compare it with a previous version
* if they are different then:
1. call **insertLegend(plot, newEntries)**
2. store the value of the **newEntries** for further comparisons